tests/src/test/java/alluxio/collections/IndexedSetConcurrencyTest.java

```diff
@@ -76,6 +78,25 @@ public long runSingleTask() {
     }
   }
 
+  private class ConcurrentAddWithCheck extends ConcurrentTask {
+    @Override
+    public long runSingleTask() {
+      long result = 0;
```

Comment:
I suggest to use `Barrier` to sync threads, so those different tasks can really sync together to start running at the same time, even hardening the test.

see https://github.com/Alluxio/alluxio/pull/3611/files